packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexPageKanbanAddButton.tsx

```diff
@@ -0,0 +1,96 @@
+import { Dropdown } from '@/ui/layout/dropdown/components/Dropdown';
+import { DropdownMenu } from '@/ui/layout/dropdown/components/DropdownMenu';
+import { DropdownMenuItemsContainer } from '@/ui/layout/dropdown/components/DropdownMenuItemsContainer';
+import styled from '@emotion/styled';
+import { useState } from 'react';
+
+import { SingleEntitySelect } from '@/object-record/relation-picker/components/SingleEntitySelect';
+
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { RecordIndexPageKanbanAddMenuItem } from '@/object-record/record-index/components/RecordIndexPageKanbanAddMenuItem';
+import { useRecordIndexPageKanbanAddButton } from '@/object-record/record-index/hooks/useRecordIndexPageKanbanAddButton';
+import { IconButton } from '@/ui/input/button/components/IconButton';
+import { IconPlus } from 'twenty-ui';
+
+const StyledDropdownMenuItemsContainer = styled(DropdownMenuItemsContainer)`
+  width: 100%;
+`;
+
+type RecordIndexPageKanbanAddButtonProps = {
+  recordIndexId: string;
+  objectNamePlural: string;
+};
+
+export const RecordIndexPageKanbanAddButton = ({
+  recordIndexId,
+  objectNamePlural,
+}: RecordIndexPageKanbanAddButtonProps) => {
+  const [isSelectingCompany, setIsSelectingCompany] = useState(false);
+
+  const {
+    dropdownId,
+    columnIds,
+    selectFieldMetadataItem,
+    isOpportunity,
+    handleOpportunityClick,
```

Comment:
Could we keep the handlers in the component ? Generally it's better to use hook for logic unrelated to component management, so instead of exporting handlers let's export functions like createOpportunity or createOpportunityWithoutCompany, etc. And then in a handleItemClick we would handle the ternary logic. 

That helps us keep business logic separated from component logic.